src/main.cairo

```diff
@@ -1,32 +1,231 @@
 %lang starknet
 from starkware.cairo.common.math import assert_nn
 from starkware.cairo.common.cairo_builtins import HashBuiltin
+from starkware.starknet.common.syscalls import get_caller_address, get_block_timestamp, get_contract_address
+from starkware.cairo.common.math import assert_not_zero, assert_le
+from starkware.cairo.common.registers import get_label_location
+
+from cairo_contracts.src.openzeppelin.upgrades.library import Proxy
+from lib.cairo_contracts.src.openzeppelin.token.erc20.IERC20 import IERC20
+
+from src.interface.naming import Naming
+from src.interface.pricing import Pricing
+from src.constants import VOTERS_LEN, VOTING_QUORUM, voters_data_start
+
+//
+// Storage
+//
+
+@storage_var
+func admin_address() -> (address: felt) {
+}
+
+@storage_var
+func naming_contract() -> (contract_address: felt) {
+}
+
+@storage_var
+func pricing_contract() -> (contract_address: felt) {
+}
+
+@storage_var
+func is_renewing(renewer: felt, domain: felt) -> (bool: felt) {
+}
 
 @storage_var
-func balance() -> (res: felt) {
+func voted_upgrade(user: felt, upgrade_id: felt, implementation_hash: felt) -> (bool: felt) {
+}
+
+@storage_var
+func blacklisted(upgrade_id: felt) -> (bool: felt) {
+}
+
+//
+// Events
+//
+
+@event
+func toggled_renewal(domain: felt, renewer: felt, value: felt) {
+}
+
+@event
+func domain_renewed(domain: felt, renewer: felt, days: felt) {
+}
+
+@event
+func voted(caller: felt, upgrade_id: felt, implementation_hash: felt, vote: felt) {
 }
 
 @external
-func increase_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-    amount: felt
+func initializer{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    admin: felt, 
+    naming_address: felt, 
+    pricing_address: felt
 ) {
-    with_attr error_message("Amount must be positive. Got: {amount}.") {
-        assert_nn(amount);
-    }
+    // Can only be called if there is no admin
+    let (current_admin) = admin_address.read();
+    assert current_admin = 0;
+
+    admin_address.write(admin);
+    naming_contract.write(naming_address);
+    pricing_contract.write(pricing_address);
 
-    let (res) = balance.read();
-    balance.write(res + amount);
+    Proxy.initializer(admin);
     return ();
 }
 
+// @notice Get the status of renewals for a domain
+// @param domain Domain to get status of
+// @param user User to get status of
 @view
-func get_balance{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}() -> (res: felt) {
-    let (res) = balance.read();
+func will_renew{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+    domain: felt, 
+    renewer: felt
+) -> (res: felt) {
+    let (res) = is_renewing.read(renewer, domain);
```

Comment:
To me the best naming here would be

Storage var `_is_renewing`
function `is_renewing`

Doing two different naming can be confusing (but it's actually the same thing).